### PR TITLE
explicity handle whitespace

### PR DESCRIPTION
**Pull request by tyrasd**
This allows to reject strings with whitespace before/after colons, before/after wildcards in pattern matches, or other situations where extra whitespace is unwanted.

For example, the following input is now correctly rejected as invalid:
* `sidewalk : left = yes` (correct: `sidewalk:left = yes`)
* `maxspeed ~ * mph` (correct: `maxspeed ~ *mph` or `maxspeed ~ *" mph"`)

Comes at the cost of slightly less readable error messages (they may include an extra _"…WS expected"_ where whitespace is allowed, see changed test cases), a slightly more complex grammar (operators need to explicitly allow whitespace before/after, e.g. `WS? '!=' WS?`) and we need to `strip` some strings in the listener to remove extra spaces where we don't want them in the output.

Overall, I'm not quite 100% sure if the downsides are worth catching the obscure-ish error cases above :shrug:  so marking as darft PR for now. Feedback welcome.

### CODE REVIEW

File: ohsome_filter_to_sql/main.py [hunk 1]
```diff
@@ -50,21 +50,30 @@ def exitString(self, ctx: ParserRuleContext):
                 result += child.getText()
             self.stack.append(result)
 
+    def exitRange_int(self, ctx):
```
Comment (matthiasschaub) [inline ohsome_filter_to_sql/main.py:53]:
```suggestion
    def exitRange_int(self, ctx):
    """Remove range brackets."""
```

File: ohsome_filter_to_sql/main.py [hunk 2]
```diff
@@ -50,21 +50,30 @@ def exitString(self, ctx: ParserRuleContext):
...
+        self.stack.append(ctx.getText().strip()[1:-1].strip())
+
+    def exitRange_dec(self, ctx):
```
Comment (matthiasschaub) [inline ohsome_filter_to_sql/main.py:56]:
```suggestion
    def exitRange_dec(self, ctx):
    """Remove range brackets."""
```

File: ohsome_filter_to_sql/main.py [hunk 3]
```diff
@@ -50,21 +50,30 @@ def exitString(self, ctx: ParserRuleContext):
...
+
     # --- methods are sorted in the same order as rules in OFL.g4
     #
     def exitExpression(self, ctx: ParserRuleContext):
         """Handle expression compositions: (), NOT, AND, OR"""
         if ctx.getChildCount() == 3:
-            if ctx.getChild(0).getText() == ("(") and ctx.getChild(2).getText() == ")":
+            if (
+                ctx.getChild(0).getText().strip() == ("(")
+                and ctx.getChild(2).getText().strip() == ")"
```
Comment (matthiasschaub) [inline ohsome_filter_to_sql/main.py:66]:
```suggestion
                ctx.getChild(0).getText().strip() == "("
                and ctx.getChild(2).getText().strip() == ")"
```

File: ohsome_filter_to_sql/main.py [hunk 4]
```diff
@@ -81,10 +90,8 @@ def exitHashtagListMatch(self, ctx: ParserRuleContext):
         values = []
         children = list(child.getText() for child in ctx.getChildren())
         # skip first part denoting "hashtag:(" and last part closing list with ")"
-        for child in children[3:-1]:
-            # skip comma in list in between brackets
-            if child == ",":
-                continue
+        # and skip commas in list in between brackets
```
Comment (matthiasschaub) [inline ohsome_filter_to_sql/main.py:93]:
```suggestion
        # and skip commas in list in between brackets (every second child)
```
